pubsub/schemas/subscribe_proto_messages.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+// [START pubsub_subscribe_proto_messages]
+import (
+	"context"
+	"fmt"
+	"io"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+	statepb "github.com/GoogleCloudPlatform/golang-samples/internal/pubsub/schemas"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+func subscribeWithProtoSchema(w io.Writer, projectID, subID, protoFile string) error {
+	// projectID := "my-project-id"
+	// subID := "my-sub"
+	// protoFile = "path/to/a/proto/schema/file(.proto)/formatted/in/protocol/buffers"
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+
+	state := &statepb.State{}
```

Comment:
maybe a comment here explaining that this is creating an instance of the message to be decoded?  First time I read this I thought this was related to pubsub state, not a message about a US state.